src/lib/cimbar_js/cimbar_zstd_js.cpp

```diff
@@ -0,0 +1,62 @@
+/* This code is subject to the terms of the Mozilla Public License, v.2.0. http://mozilla.org/MPL/2.0/. */
+#include "cimbar_zstd_js.h"
+
+#include "compression/zstd_decompressor.h"
+
+#include <memory>
+#include <sstream>
+
+// wasm api will be something like
+// zstd_init_stream(bytes) -> constructor, reads skippable frame at start? (needs to for ofstream case)
+// get_filename(buff, size) -> cached in wasmjs api from constructor call?
+// get_bytes(buff, size) -> "decompress_chunk()/write()"
+
+namespace {
+	// maybe a map eventually
+	std::unique_ptr<cimbar::zstd_decompressor<std::stringstream>> _dec;
+
+}
+
+// zstd_decompressor constructor, reads skippable frame at start? (needs to for ofstream case)
+// if we make _dec a map, this'll also take the uin32_t id we got from the decoder sink, probably
+int cimbarz_init_decompress(unsigned char* buffer, unsigned size)
+{
+	if (_dec)
+		_dec.reset();
```

Comment:
You can *try* to decompress multiple files in parallel, but you won't get very far...